core/src/types.ts

```diff
@@ -269,22 +269,27 @@ export declare interface DebtAuctionTransaction extends DebtAuctionEnriched, Com
 
 type CollateralType = CollateralConfig['title'];
 
+export declare interface CollateralLiquidationLimit {
+    currentCollateralDebtDai: BigNumber;
+    maximumCollateralDebtDai: BigNumber;
+}
 export declare interface LiquidationLimits {
     maximumProtocolDebtDai: BigNumber;
-    maximumCollateralDebtDai: BigNumber;
     currentProtocolDebtDai: BigNumber;
-    currentCollateralDebtDai: BigNumber;
+    collateralDebts: Record<CollateralType, CollateralLiquidationLimit>
 }
```

Comment:
As suggested, I don't think this is a good idea to complicate this object and cache it ourselves. This can be simplified via using `memoizee` over `getCollateralLiquidationLimits(network, collateralType)` if `LiquidationLimits` will become part of the `VaultTransaction`

Reply:
answered in the q linked.